### PR TITLE
Fix: Deploying Port-less Container Image Fails

### PR DESCRIPTION
**Pull request by gijohn**
Fixes: https://jira.coreos.com/browse/ODC-2060

### CODE REVIEW

File: frontend/packages/dev-console/src/utils/shared-submit-utils.ts [hunk 1]
```diff
@@ -10,6 +10,12 @@ export const annotations = {
 
 export const dryRunOpt = { queryParams: { dryRun: 'All' } };
 
+const isDeployImageFormData = (
+  formData: DeployImageFormData | GitImportFormData,
+): formData is DeployImageFormData => {
+  return (formData as DeployImageFormData).isi !== undefined;
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/utils/shared-submit-utils.ts]: May be write this as:
```suggestion
  return !_.isEmpty((formData as DeployImageFormData).isi);

Comment (christianvogt) [reply frontend/packages/dev-console/src/utils/shared-submit-utils.ts]: or `'isi' in formData` or `formData.hasOwnProperty('isis')` ?

**Comment by andrewballantyne** [comment]
LGTM - Tested it and it seems to have nicely solved the issue.

**Comment by christianvogt** [comment]
@gijohn please add a unit test

**Comment by gijohn** [comment]
> @gijohn please add a unit test

@christianvogt I have added another validation. Previously the test was written for a wrong logic. I have corrected that too.

**Pull request merged**